crates/test-programs/src/bin/preview2_stream_pollable_correct.rs

```diff
@@ -1,11 +1,27 @@
-use test_programs::wasi::cli::stdin;
-use test_programs::wasi::io::streams;
+use test_programs::wasi::cli::{stdin, stdout};
 
 fn main() {
-    let stdin: streams::InputStream = stdin::get_stdin();
+    let stdin = stdin::get_stdin();
     let stdin_pollable = stdin.subscribe();
     stdin_pollable.block();
     assert!(stdin_pollable.ready(), "after blocking, pollable is ready");
     drop(stdin_pollable);
     drop(stdin);
+
+    // Pollables can be used many times over their lifetime
+    let stdout = stdout::get_stdout();
+    let stdout_pollable = stdout.subscribe();
+
+    let chunk = [b'a'; 50];
+    for _ in 1..10 {
+        stdout_pollable.block();
+        assert!(stdout_pollable.ready(), "after blocking, pollable is ready");
+
+        let n = stdout.check_write().unwrap() as usize;
```

Comment:
Perhaps this should write until check write doesn't allow anymore writing to ensure it's hitting backpressure?

The issue here in JCO is that we don't have backpressure for stdout since Node.js doesn't support it (https://nodejs.org/docs/latest/api/process.html#a-note-on-process-io).

Reply:
I'm working separately on a test that pipes stdout to stdin of another instantiation of the same component, and I think will give us a lot more control over how this test works. I'll ping you when it's ready :)